crates/file_finder/src/file_finder.rs

```diff
@@ -715,14 +788,17 @@ impl FileFinderDelegate {
 
     /// Skips first history match (that is displayed topmost) if it's currently opened.
     fn calculate_selected_index(&self) -> usize {
-        if let Some(Match::History(path, _)) = self.matches.get(0) {
-            if Some(path) == self.currently_opened_path.as_ref() {
-                let elements_after_first = self.matches.len() - 1;
-                if elements_after_first > 0 {
-                    return 1;
+        if let Some(m) = self.matches.get(0) {
```

Comment:
NIT: we can do `if let Some(Match::History(entry, _)) = ...` and reduce the level of nesting.